pkg/kubectl/cmd/set/helper.go

```diff
@@ -120,9 +120,12 @@ func CalculatePatches(infos []*resource.Info, encoder runtime.Encoder, mutateFn
 	for _, info := range infos {
 		patch := &Patch{Info: info}
 		patch.Before, patch.Err = runtime.Encode(encoder, info.Object)
+		if patch.Err != nil {
+			continue
+		}
 
 		ok, err := mutateFn(info)
-		if !ok {
+		if !ok && err == nil {
```

Comment:
@kargakis talking about this (the fix above is also needed as we swallowing the encoding errors (the `patch.Err` is overrided by the code below.... ))